src/session.c

```diff
@@ -56,52 +56,79 @@
 
 extern struct nc_server_opts server_opts;
 
-int
-nc_gettimespec_mono(struct timespec *ts)
+/* ts1 < ts2 -> +, ts1 > ts2 -> -, returns milliseconds */
+int32_t
+nc_difftimespec(const struct timespec *ts1, const struct timespec *ts2)
 {
-#ifdef CLOCK_MONOTONIC_RAW
-    return clock_gettime(CLOCK_MONOTONIC_RAW, ts);
-#elif defined (CLOCK_MONOTONIC)
-    return clock_gettime(CLOCK_MONOTONIC, ts);
-#else
-    /* no monotonic clock available, return realtime */
-    return nc_gettimespec_real(ts);
-#endif
+    int64_t nsec_diff = 0;
+
+    nsec_diff += (((int64_t)ts2->tv_sec) - ((int64_t)ts1->tv_sec)) * 1000000000L;
+    nsec_diff += ((int64_t)ts2->tv_nsec) - ((int64_t)ts1->tv_nsec);
+
+    return nsec_diff ? nsec_diff / 1000000L : 0;
```

Comment:
A trivial example of them being equal is calling this function with the exact same arguments :). But the point is that the check appears to be redundant; there is no harm dividing a zero number by a large constant. It will return the exact same zero as the `: 0` part of the expression.